File: Java Advanced/Java OOP/Encapsulation - Lab/SalaryIncrease02/Person.java
package SalaryIncrease02;

public class Person {

    private String firstName;
    private String lastName;
    private int age;
    private double salary;


    public Person(String firstName, String lastName, int age, double salary) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.salary = salary;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public int getAge() {
        return age;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public void increaseSalary(double incrementPercentage) {
        if(this.age < 30) {
            incrementPercentage /= 2;
        }
        incrementPercentage /= 100;

     this.salary += this.salary * incrementPercentage;
        System.out.printf("%s %s gets %s leva", firstName, lastName, salary);
    }

    @Override
    public String toString() {
        return String.format("%s %s is %d years old.", firstName, lastName, age);
    }
}

File: JavaScript Advanced/Prototype/person.js
class Person {
    constructor(firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    get fullName() {
        return `${this.firstName} ${this.lastName}`;
    }

    set fullName(value) {
        let [firstName, lastName] = value.split(' ');

        if (firstName && lastName) {
            this.lastName = lastName;
            this.firstName = firstName;
        }
    }
}